### ISSUE TITLE
[Support]: Retain Days & Deletion

### ISSUE THREAD

**Issue by D3v01dZA** [blakeblackshear/frigate#2003]
### Describe the problem you are having

I migrated up to 0.9.1 as it was released (thank you, this project is seriously fantastic and I've been trying to contribute but am running into issues getting everything working locally).

Now I have a couple questions about retain_days because it isn't clear if its honoring that or not to me. I have recording retain set to 31 days, however there are still recordings for September 07 showing up on the Media Browser in Home Assistant. Clicking on these recordings fails to load them, but that also happens to recordings within the 31 day window that were recorded before I migrated. I didn't notice anything in the release notes about this or what I should do.

So the first thing I'd like help with is understanding how I can remove the space of the extra retain_days recordings, the Web UI shows only the days since I updated, Home Assistant shows up to September 07 and the files in my recordings directory go to September 07. Can I simply delete the extra files beyond the 31 day period, or is it stored in a database somewhere as well (I saw mentions of referencing clips from recordings so I feel like it would be).

Also it may help to add something to the documentation about this, when does retain_days take effect, does it delete hour by hour as its saving the recordings (in the new version mine goes by hours instead of minutes)? Does it run a job overnight? How do we clean up when something goes wrong?

Thanks again for your help and this awesome project!

### Version

0.9.1-800f33e

### Frigate config file

```yaml
mqtt:
  - redacted

detectors:
  cpu:
    type: cpu
    num_threads: 6

ffmpeg:
  input_args:
    - -c:v
    - h264_cuvid

record:
  enabled: true
  retain_days: 31
  events:
    retain:
      default: 31

detect:
  width: 1280
  height: 720
  fps: 5

snapshots:
  enabled: true 

cameras:
  doorbell_camera:
    ffmpeg:
      inputs:
        - path: redacted
          roles:
            - rtmp
            - detect
            - record
    motion:
      mask:
        - 0,1080,759,1080,766,1053,764,1024,733,965,657,942,692,877,879,932,955,983,1010,1080,1920,1080,1920,0,0,0
  trainer_camera:
    ffmpeg:
      inputs:
        - path: redacted
          roles:
            - rtmp
            - detect
            - record
    objects:
      track:
        - person
        - cat
        - dog
    motion:
      mask:
        - 2494,32,2490,142,1742,150,1742,33
  patio_camera:
    ffmpeg:
      inputs:
        - path: redacted
          roles:
            - rtmp
            - detect
            - record
    objects:
      track:
        - person
        - cat
        - dog
    motion:
      mask:
        - 2494,32,2490,142,1742,150,1742,33
```

### Relevant log output

```shell
Not much interesting here, but I added it

Attaching to frigate
frigate    | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
frigate    | [s6-init] ensuring user provided files have correct perms...exited 0.
frigate    | [fix-attrs.d] applying ownership & permissions fixes...
frigate    | [fix-attrs.d] done.
frigate    | [cont-init.d] executing container initialization scripts...
frigate    | [cont-init.d] done.
frigate    | [services.d] starting services
frigate    | [services.d] done.
frigate    | [2021-10-13 09:46:26] frigate.app                    INFO    : Starting Frigate (0.9.1-800f33e)
frigate    | Starting migrations
frigate    | [2021-10-13 09:46:26] peewee_migrate                 INFO    : Starting migrations
frigate    | There is nothing to migrate
frigate    | [2021-10-13 09:46:26] peewee_migrate                 INFO    : There is nothing to migrate
frigate    | [2021-10-13 09:46:27] frigate.mqtt                   INFO    : MQTT connected
frigate    | [2021-10-13 09:46:27] detector.cpu                   INFO    : Starting detection process: 256
frigate    | [2021-10-13 09:46:27] frigate.app                    INFO    : Output process started: 258
frigate    | [2021-10-13 09:46:27] frigate.app                    INFO    : Camera processor started for doorbell_camera: 261
frigate    | [2021-10-13 09:46:27] frigate.edgetpu                WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
frigate    | [2021-10-13 09:46:27] frigate.app                    INFO    : Camera processor started for trainer_camera: 263
frigate    | [2021-10-13 09:46:27] ws4py                          INFO    : Using epoll
frigate    | [2021-10-13 09:46:27] frigate.app                    INFO    : Camera processor started for patio_camera: 265
frigate    | [2021-10-13 09:46:27] frigate.app                    INFO    : Capture process started for doorbell_camera: 266
frigate    | [2021-10-13 09:46:27] frigate.app                    INFO    : Capture process started for trainer_camera: 270
frigate    | [2021-10-13 09:46:27] frigate.app                    INFO    : Capture process started for patio_camera: 276
frigate    | [2021-10-13 09:46:27] ws4py                          INFO    : Using epoll
frigate    | [2021-10-13 09:46:40] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:55252]
frigate    | [2021-10-13 09:53:06] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:55252]
frigate    | [2021-10-13 09:53:06] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:58232]
frigate    | [2021-10-13 09:53:07] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:58232]
frigate    | [2021-10-13 09:53:07] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:58258]
frigate    | [2021-10-13 09:53:08] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:58258]
frigate    | [2021-10-13 09:54:12] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:47154]
frigate    | [2021-10-13 09:54:14] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:58990]
frigate    | [2021-10-13 09:55:12] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:8082 | Remote => 127.0.0.1:58990]
frigate    | [2021-10-13 09:55:12] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:47154]
frigate    | [2021-10-13 10:00:13] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:49424]
frigate    | [2021-10-13 10:01:37] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:49424]
```

### FFprobe output from your camera

```shell
Not relevant
```

### Frigate stats

```json
{"detection_fps":1.5,"detectors":{"cpu":{"detection_start":0.0,"inference_speed":30.91,"pid":256}},"doorbell_camera":{"camera_fps":5.1,"capture_pid":266,"detection_fps":0.0,"pid":261,"process_fps":5.1,"skipped_fps":0.0},"patio_camera":{"camera_fps":5.0,"capture_pid":276,"detection_fps":0.0,"pid":265,"process_fps":5.0,"skipped_fps":0.0},"service":{"storage":{"/dev/shm":{"free":2142.2,"mount_type":"tmpfs","total":2147.5,"used":5.3},"/media/frigate/clips":{"free":6358711.7,"mount_type":"ext4","total":7937383.5,"used":1178577.6},"/media/frigate/recordings":{"free":6358711.7,"mount_type":"ext4","total":7937383.5,"used":1178577.6},"/tmp/cache":{"free":3989.5,"mount_type":"tmpfs","total":4000.0,"used":10.5}},"uptime":1298,"version":"0.9.1-800f33e"},"trainer_camera":{"camera_fps":5.1,"capture_pid":270,"detection_fps":1.5,"pid":263,"process_fps":5.1,"skipped_fps":0.0}}
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Network connection

Mixed

### Camera make and model

Amcrest

**Comment by brgaulin** [comment]
I was experiencing the same. After upgrading to the new version, I ran out of disk space as the older recordings didn't rotate out. It was easy enough to manually delete them, will keep an eye on it =)

**Comment by lindsaymarkward** [comment]
I came looking for this - old recordings not being deleted as I'm running out of disk space every so often.
I've reduced the number of days to retain, but it seems it still keeps old recordings.
When I restart the Raspberry Pi, I get my disk space back, but I don't see it happening regularly without a restart.

> I think I found the problem. It wasn't working properly.

Has this problem been solved yet, or should I wait for a future release?
Thanks!
